fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/database/core/CoreEntity.java

```diff
@@ -0,0 +1,100 @@
+package com.github.dedis.popstellar.repository.database.core;
+
+import androidx.annotation.NonNull;
+import androidx.room.*;
+
+import com.github.dedis.popstellar.model.objects.Channel;
+
+import java.util.*;
+
+@Entity(tableName = "core")
+public class CoreEntity {
+
+  // The id is used to keep always at most one entry in the database
+  @ColumnInfo(name = "id")
+  @PrimaryKey
+  private final int id;
+
+  @ColumnInfo(name = "server_address")
+  @NonNull
+  private final String serverAddress;
+
+  @ColumnInfo(name = "wallet_seed")
+  @NonNull
+  private final List<String> walletSeed;
+
+  @ColumnInfo(name = "subscription")
+  @NonNull
+  private final Set<Channel> subscriptions;
+
+  public CoreEntity(
+      int id,
+      @NonNull String serverAddress,
+      @NonNull List<String> walletSeed,
+      @NonNull Set<Channel> subscriptions) {
+    this.id = id;
+    this.serverAddress = serverAddress;
+    this.walletSeed = walletSeed;
+    this.subscriptions = subscriptions;
+  }
+
+  public int getId() {
+    return id;
+  }
+
+  @NonNull
+  public String getServerAddress() {
+    return serverAddress;
+  }
+
+  @NonNull
+  public List<String> getWalletSeed() {
+    return walletSeed;
+  }
+
+  @NonNull
+  public Set<Channel> getSubscriptions() {
+    return subscriptions;
+  }
+
+  public String[] getWalletSeedArray() {
+    return walletSeed.toArray(new String[0]);
+  }
+
+  public Set<Channel> getSubscriptionsCopy() {
+    return new HashSet<>(subscriptions);
+  }
```

Comment:
Could you copy the arrays ? Also no need to add the suffix copy for the one where you did